### PR TITLE
Update new tab page to use FrecentHistory

### PR DESCRIPTION
**Pull request by k88hudson**
* Updates `NewTabPage` to use `FrecentHistory` instead of `History`
* Fixes bug where `FRECENT_LINKs_REQUEST` wasn't being listened for (d452ffb)
* Adjusts `getFrecentLinks` query to prioritize bookmarks (34bb017)
* Adds local sorting by date in `TopActivity`

### CODE REVIEW

File: content-test/selectors/selectors.test.js [hunk 1]
```diff
@@ -117,5 +111,24 @@ describe("selectors", () => {
...
+      state = selectNewTabSites({
+        TopSites: {rows: []},
+        Spotlight: {rows: []},
+        FrecentHistory: {rows: [
+          {url: "foo.com", lastVisitDate: 1},
+          {url: "bar.com", lastVisitDate: 4},
+          {url: "baz.com", lastVisitDate: 3}
+        ]}
+      });
+      console.log(state.TopActivity.rows);
```
Comment (pdehaan) [inline content-test/selectors/selectors.test.js]: Still need the `console.log()` here?

Comment (k88hudson) [inline content-test/selectors/selectors.test.js]: oops!

Comment (pdehaan) [inline content-test/selectors/selectors.test.js]: FWIW, Travis-CI caught it, but it's currently configured as an ESLint warning (not an error), so you'll only catch it if you are looking through Travis logs... 

https://travis-ci.org/mozilla/activity-streams/builds/114623632#L278

Let me know if you want to convert any ESLint rules from warnings=>errors or warnings=>ignore (or any other combination).

Comment (k88hudson) [inline content-test/selectors/selectors.test.js]: I think i prefer the warnings, although we should try to get the existing ones cleaned up (it's hard to see new ones)

Comment (k88hudson) [inline content-test/selectors/selectors.test.js]: Maybe we want to remove the TODO one for now... what do you think?

Comment (pdehaan) [inline content-test/selectors/selectors.test.js]: Yeah, I can do a quick PR tomorrow to hide the `no-warning-comments` (TODO) rule, and maybe hide the `no-shadow` rule for now too. That would put us at 0 warnings, making anything new fairly visible (or as visible as warnings can be).

**Comment by k88hudson** [comment]
R? @rlr or whoever

**Comment by rlr** [comment]
Starting with a fresh profile, I can't seem to get anything to show up in spotlight on any of the pages. Or on top activity of the new tab page.

**Comment by rlr** [comment]
(I visited some pages. Bookmarked some. etc.)

File: lib/PlacesProvider.js [hunk 2]
```diff
@@ -222,8 +222,8 @@ Links.prototype = {
     }
 
     // setup order by clause
-    let orderbyClause = (order === "frecency") ?
-                    "ORDER BY frecency DESC, lastVisitDate DESC, url" :
+    let orderbyClause = (order === "bookmarksFrecency") ?
+                    "ORDER BY bookmarkDateCreated DESC, frecency DESC, lastVisitDate DESC, url" :
```
Comment (oyiptong) [inline lib/PlacesProvider.js]: Comment for the function parameters has been obsoleted.

We'll also want tests for ordering.

**Comment by pdehaan** [comment]
Not working for me locally.

new_tab_and_3__npm_run_once__sh__and_1__npm_run_once__node_
**Figure 1:** "Top Sites" is populated, but not "Spotlight" or "Top Activity".

---

new_tab
**Figure 2:** "Spotlight" results still empty on Timeline page, but the Activity Stream has my 18 recent history items.

**Comment by k88hudson** [comment]
@rlr ok should be working now, I mixed up on/off by accident

File: lib/PlacesProvider.js (hunk 2 above)
Comment (k88hudson) [inline lib/PlacesProvider.js]: :+1:

**Comment by rlr** [comment]
works fantastic now. r=me

**Comment by oyiptong** [comment]
r+ on the addon side

**Pull request merged**